File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.py
class Solution(object):
    def minimumDeletions(self, s):
        """
        :type s: str
        :rtype: int
        """

        min_del = 0
        count_b = 0

        for char in s:
            if char == 'a':
                min_del = min(min_del + 1, count_b)
            else:
                count_b = count_b + 1

            min_del = min(min_del, count_b)

        return min_del
        

File: 1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        long long min_deletion = 0;
        long long count_b = 0;

        for (char c:s){
            if (c == 'a'){
                min_deletion = min(min_deletion+1, count_b);
            }
            else {
                count_b += 1;
            }
        }

        return (int)min_deletion;
    }
};